src/builtin_disown.cpp

```diff
@@ -101,9 +101,9 @@ int builtin_disown(parser_t &parser, io_streams_t &streams, wchar_t **argv) {
         }
 
         // Disown all target jobs
-        for (const auto &j : jobs) {
-            retval |= disown_job(cmd, parser, streams, j);
-        }
+        return std::accumulate(jobs.begin(), jobs.end(), retval, [&](int r, job_t *j) {
+            return r || disown_job(cmd, parser, streams, j);
```

Comment:
Looks like you have uncovered a bug in this linter - this should use bitwise "or" to always disown all jobs.

```cpp
#include <stdio.h>
#include <vector>
#include <numeric>

int main() {
	std::vector<int> vs = {0, 1, 2};

	int r1 = 0;
	for (const auto &v : vs)
		r1 |= v;

	int r2 = std::accumulate(vs.begin(), vs.end(), 0, [&](int r, int v) {
		return r || v; // should be r | v
	});

	printf("%d\n", r1); // prints 3
	printf("%d\n", r2); // prints 1
}
```

Reply:
Good catch.

All the changes were done manually. This was a mistake on my part.